snuba/clickhouse/formatter/expression.py

```diff
@@ -88,7 +89,16 @@ def visit_column(self, exp: Column) -> str:
             ret_unescaped.append(exp.table_name or "")
             ret.append(".")
             ret_unescaped.append(".")
-        ret.append(escape_identifier(exp.column_name) or "")
+            # If there is a table name and the column name itself is an alias,
+            # then we should apply alias escaping rules to the
+            # column name since the alias column names could have "." in them
+            # in cases like selected columns of JOINS
+            if ALIAS_PREFIX in exp.column_name:
```

Comment:
Not sure I get why we look for the column name to be an alias. 
Cannot we just escape the column name if it contains a `.` ? Or just always escape?

Relying on the alias prefix seems really fragile. Aliasing and formatting are independent features, if we make formatter rely on the prefix we make them dependent on each other in a subtile and easy to break way.
Also if I understand the problem correctly, the issue is with the fact that the column name contains a `.` without being a nested column instead of being an alias.

Reply:
I was using alias since i didn't want to escape nested column names since they were working even if there is a table name. But I tried out queries with columns as "table_name.`tags.keys`" and they work as well. So I now escape based on nested columns. 

I am not always escaping since its more intuitive to have "table_name.column_name" than "table_name.`column_name`" when the columns are not nested.

Reply:
> I am not always escaping since its more intuitive to have "table_name.column_name" than "table_name.column_name" when the columns are not nested.

That is true, but between a little more verbosity and having the formatter depending on a concept that is part of the parser (which is likely the cause of the circular dependency as well), I would take the verbosity.
The formatter and the parser have no architectural reason to be dependent on each other. They only have a common contract in the query representation.

Reply:
I feel the refactoring in addition to the bug fix was making the PR harder to understand. I have removed the refactoring from this PR so that its easier to reason about the change. 

Since there is no circular dependency between the parser and formatter, I kept the logic of checking "." in column names in the formatter. So the Query would be "table_name.``column_name``" only if the column name has a "." in it